### PR TITLE
Added a few instructions for conditional execution.

### PR DESCRIPTION
**Pull request by pakls**
SUB is used as CMP.

### CODE REVIEW

File: opcode.c [hunk 1]
```diff
@@ -46,3 +46,4 @@ void hook_opcodes(vm_env *env)
     vm_hook_opcode_handler(env, OP_ADD, add_impl);
     vm_hook_opcode_handler(env, OP_SUB, sub_impl);
 }
+
```
Comment (jserv) [inline opcode.c]: Don't append extra line here.

File: vm.c [hunk 2]
```diff
@@ -28,6 +28,23 @@
     pc = n;        \
     goto *labels[OPCODE.opcode]
 
+#define VM_J_MACRO(cond)                                         \
```
Comment (jserv) [inline vm.c]: Would you think it is better to call it `VM_J_TYPE_INST`?

File: vm.c [hunk 3]
```diff
@@ -130,12 +147,18 @@ void vm_run(vm_env *env)
     size_t pc = 0;
     BEGIN_OPCODES;
 
-    OP(ADD) : VM_CALL_HANDLER();
-    OP(SUB) : VM_CALL_HANDLER();
+    OP(ADD)   : VM_CALL_HANDLER();
```
Comment (jserv) [inline vm.c]: Don't add extra blank. Use `clang-format -i` to indent.

Comment (mlouielu) [inline vm.c]: This should not add here, isn't related to this PR.

File: vm.c [hunk 4]
```diff
@@ -50,15 +67,15 @@
 /* OPCODE impl max size */
 #define OPCODE_IMPL_MAX_SIZE 256
 
-typedef struct __vm_env {
+struct __vm_env {
```
Comment (mlouielu) [inline vm.c]: Please revert this change. This isn't related to this PR.

File: tests/jcond.s [hunk 5]
```diff
@@ -0,0 +1,31 @@
+jmp #5
+print ^cond_jump_insts_ok
+halt
+print ^cond_jump_insts_ng
+halt
+sub $1 $2 #3
+jz  #3 #3
```
Comment (jserv) [inline tests/jcond.s]: You can add some comments starting with `;`.

**Comment by mlouielu** [comment]
Also, you don't need to rebase or squash to addressed each review changed. GitHub can squash at the end of the review when PR is accepted. (@jserv: FYI,  the merged button have a squash and merged function can use)

File: vm.c (hunk 4 above)
Comment (pakls) [inline vm.c]: Should be sent in a separate PR ?

**Comment by jserv** [comment]
@mlouielu , I knew there is a merge + squash button, but it is slightly difficult for me to edit messages in consistent manner.

File: vm.c (hunk 4 above)
Comment (jserv) [inline vm.c]: Yes, please submit another pull request.

Comment (mlouielu) [inline vm.c]: Yes, it should. The main reason to separate is because your PR is focus on adding new J-series operator, not "Fix duplicate typedef". 

When in future someone trying to investigate in git log, it will be a mess if irrelevant changed is in the commit.

**Comment by mlouielu** [comment]
@jserv, yep, that is somehow hard to use. But if using git rebase and force push to GitHub, the review comment will be a mess. You can see at the top of your three comments are collapsed and labeling as "Show outdated".

Which method do you more prefer on GitHub?

**Comment by jserv** [comment]
Essential code review messages should take place in conversations rather than inlined.
At present, I prefer t ask contributors to squash and reorder Git commits.
